### PR TITLE
Styling of preferences screens

### PR DESCRIPTION
**Pull request by nallath**
The materials page isn't styled yet but noticed that i misunderstood the scope of the ticket and went a bit overbroad. 

CURA-8949 
https://github.com/Ultimaker/Uranium/pull/782

### CODE REVIEW

**Review by casperlamboo — CHANGES_REQUESTED** [review]
Code looks good. these new preference pages look so much better then before, great job @lorenzoromagnoli !

While component testing i did find some blemishes

There is something weird going on with the dropdownbox in the settings visibility page there is no left blue border when hovering
I think that "check all" checkbox should be a `UM.Checkbox` don't think that it is right now.
Screenshot from 2022-02-19 11-30-57

The italic text "you will need to restart" should be located under the checkbox.
Screenshot from 2022-02-19 11-32-50

**Comment by fieldOfView** [comment]
What I think is odd about this redesign is that it seems to mix the Cura theme and the system theme. I would think that this move to QtQuick 2 is a good opportunity to move to a full Cura theme solution. That would reduce the maintenance of GUI elements (eg the ever-lurking issue of commas and periods in number fields, tabbing through fields, etc) and it would solve the issue of MachineAction dialogs having to work in both the Cura-themed first-run wizard and the system-themed preferences dialogs.

**Comment by casperlamboo** [comment]
@fieldOfView we are slowly migrating to having everything upgraded to qtcontrols2/qt6. But due to the huge amount of components this takes a while. But. We're getting there

File: resources/qml/Preferences/GeneralPage.qml [hunk 1]
```diff
@@ -821,15 +840,15 @@ UM.PreferencesPage
...
+                UM.CheckBox
                 {
                     id: sendDataCheckbox
                     text: catalog.i18nc("@option:check","Send (anonymous) print information")
                     checked: boolCheck(UM.Preferences.getValue("info/send_slice_info"))
                     onCheckedChanged: UM.Preferences.setValue("info/send_slice_info", checked)
                 }
 
-                Button
+                Cura.SecondaryButton
```
[the remark this answers is not in the document]
Comment (nallath) [reply resources/qml/Preferences/GeneralPage.qml:851]: Yeah, but the design also shows it as being an external link, but its not. I believe that it's a mistake in the design (It was misinterpreted as an external page but it's actually a popup)

**Comment by nallath** [comment]
We are going to move over _all_ system styled stuff and completely move over to custom (and hopefully, consistently) styled items.

File: resources/qml/Preferences/MachinesPage.qml [hunk 2]
```diff
@@ -152,17 +128,39 @@ UM.ManagementPage
...
-            object: base.currentItem && base.currentItem.name ? base.currentItem.name : "";
+            id: renameDialog
+            object: base.currentItem && base.currentItem.name ? base.currentItem.name : ""
             property var machine_name_validator: Cura.MachineNameValidator { }
-            validName: renameDialog.newName.match(renameDialog.machine_name_validator.machineNameRegex) != null;
+            validName: renameDialog.newName.match(renameDialog.machine_name_validator.machineNameRegex) != null
             onAccepted:
             {
-                Cura.MachineManager.renameMachine(base.currentItem.id, newName.trim());
+                Cura.MachineManager.renameMachine(base.currentItem.id, newName.trim())
```
[the remark this answers is not in the document]
Comment (nallath) [reply resources/qml/Preferences/MachinesPage.qml:137]: Well, neither need it and in this case it was me trying to at least make it consistent with the next line.

File: resources/qml/Preferences/GeneralPage.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lorenzoromagnoli) [reply resources/qml/Preferences/GeneralPage.qml:851]: ah, this is probably my bad. can we the learn more with this? 

Screen Shot 2022-02-21 at 09 52 41

notice the normal and hover state

File: resources/qml/Preferences/GeneralPage.qml [hunk 3]
```diff
@@ -125,6 +125,14 @@ UM.PreferencesPage
         pluginNotificationsUpdateCheckbox.checked = boolCheck(UM.Preferences.getValue("info/automatic_plugin_update_check"))
     }
 
+    buttons: [
+        Cura.SecondaryButton
+        {
+            id: addMenuButton
```
[the remark this answers is not in the document]
Comment (nallath) [reply resources/qml/Preferences/GeneralPage.qml]: Whoops, i must have copypasted it and forgot to remove that.

File: resources/qml/Preferences/GeneralPage.qml [hunk 4]
```diff
@@ -100,14 +100,14 @@ UM.PreferencesPage
...
+        //UM.Preferences.resetPreference("view/top_layer_count");
+        //topLayerCountCheckbox.checked = boolCheck(UM.Preferences.getValue("view/top_layer_count"))
         UM.Preferences.resetPreference("general/restore_window_geometry")
         restoreWindowPositionCheckbox.checked = boolCheck(UM.Preferences.getValue("general/restore_window_geometry"))
 
         UM.Preferences.resetPreference("general/camera_perspective_mode")
-        var defaultCameraMode = UM.Preferences.getValue("general/camera_perspective_mode")
-        setDefaultCameraMode(defaultCameraMode)
+        //var defaultCameraMode = UM.Preferences.getValue("general/camera_perspective_mode")
+//        /setDefaultCameraMode(defaultCameraMode)
```
[the remark this answers is not in the document]
Comment (nallath) [reply resources/qml/Preferences/GeneralPage.qml:110]: I'm not entirely sure. They seem to be referencing things that aren't there anymore, but I don't know what happened to them.

**Pull request merged**